app/assets/javascripts/3dbio_viewer/src/webapp/components/model-upload/ModelUpload.tsx

```diff
@@ -38,39 +41,42 @@ export const ModelUpload: React.FC<ModelUploadProps> = React.memo(props => {
                 structureFile,
                 annotationsFile: getFile(annotationFileRef),
             };
-            return compositionRoot.uploadAtomicStructure(uploadParams).run(result => {
-                setAtomicStructure(result);
-                openUploadConfirmation();
-            }, console.error);
+            setOpen(true);
+            return compositionRoot.uploadAtomicStructure(uploadParams).run(
+                result => {
+                    setAtomicStructure(result);
+                    setOpen(false);
+                    openUploadConfirmation();
+                },
+                error => {
+                    setOpen(false);
+                    return error;
```

Comment:
The return values of `Future.run` callbacks are ignored, so no need to return anything. It's the only point we work with side-effects. I've added `setError(error.message)`, does it make sense?